### ISSUE TITLE
Minimise to tray not working on Wayland, since Wayland has no concept of "iconised" windows (Upstream issue bug 1872790, unfixable?)

### ISSUE THREAD

**Issue by Alex9001** [Betterbird/thunderbird-patches#279]
When using Wayland on Plasma 6 Betterbird does not have a tray icon showing and does not minimize to tray. I have minimize to tray enabled. 

![image](https://github.com/Betterbird/thunderbird-patches/assets/2770710/96405f28-b642-4543-8934-ce12d17e326b)

Works as it should in X11.

OS: Artix Linux
Kernel: 6.7.9-zen1-1-zen 
DE: Plasma 6.0.1 
WM: kwin

**Comment by Betterbird** [comment]
Copy of https://github.com/Betterbird/thunderbird-patches/issues/111#issuecomment-1935561745 follows:

---

If I see it correctly, Wayland isn't fully supported by the Mozilla platform, see all the "Depends on" in this meta bug:
https://bugzilla.mozilla.org/show_bug.cgi?id=635134

That the window doesn't minimise is due to the fact that internally a "sizemodechange" is not received. We debugged this with another BB user, who in the end reported this upstream bug:
https://bugzilla.mozilla.org/show_bug.cgi?id=1872790

You can repeat the steps in the description of this bug, that is, paste
top.window.addEventListener("sizemodechange", () => console.log("sizemodechange received"))
into the error console and see what gets logged when minimising the window.

---

You will find that nothing gets logged, hence the app doesn't receive the signal that it was minimised, so it can't minimise itself to the tray. Sorry, blame it on the Mozilla platform.

**Comment by Betterbird** [comment]
I'm reopening this, so it's easier to find.

**Comment by Betterbird** [comment]
Please note all the fine debugging we've done on the issue, see: https://github.com/Betterbird/thunderbird-patches/issues/377#issuecomment-2510216165.

TL;DR: Wayland doesn't report windows being minimised or iconised.

**Comment by Betterbird** [comment]
Please try this build:
https://www.betterbird.eu/downloads/LinuxArchive/betterbird-128.5.0esr-bb18-latest-build.en-US.linux-x86_64.tar.bz2

**Comment by Phr33d0m** [comment]
The current build in the AUR[1] stopped getting minimized to the tray (I was on 115.5 before, which worked just fine).

I can confirm that the linked build above works just fine for me on Wayland + Plasma 6. After enabling "minimize to tray" in the Settings, minimizing works as expected.

[1] aur/betterbird-bin 128.5.0esr-1 (+174 15.57) (Installed)

**Comment by Betterbird** [comment]
No problems when restoring the window to "normal" (not maximised) size? See discussion at https://github.com/Betterbird/thunderbird-patches/issues/377#issuecomment-2512669145.

**Comment by Betterbird** [comment]
> I was on 115.18 before, which worked just fine

No, it didn't, see long discussion in Issue #377. Turned out that 115 never ran under Wayland (https://github.com/Betterbird/thunderbird-patches/issues/377#issuecomment-2510253070 and above).

**Comment by Betterbird** [comment]
Build updated:
https://www.betterbird.eu/downloads/LinuxArchive/betterbird-128.5.0esr-bb18-latest-build3.en-US.linux-x86_64.tar.bz2

We're not going to post further updates, just use the latest "latest".

**Comment by PACHAKUTlQ** [comment]
So is it fixed? Mine still doesn't work.

**Comment by Betterbird** [comment]
Yes, it was implemented, however, restoring the window doesn't quite work, see Issue #399.

**Comment by MixusMinimax** [comment]
Alternatively, why don't we just have a dbus system-tray icon that is _always_ there, just like in almost every other program? If you open discord for example, the app tray icon is there immediately. And then it doesn't matter whether you minimize, close, hide, do whatever with the window, you will always see "new message" counters in the tray until you fully exit the program by for example right clicking the tray icon and clicking "exit". That is how all programs that I know work, except for thunderbird.

**Comment by Betterbird** [comment]
There is an option to always show the system tray icon, I use it myself.
